src/main/java/com/commercetools/sync/taxcategories/utils/TaxRatesUpdateActionUtils.java

```diff
@@ -0,0 +1,234 @@
+                    .map(taxRateDraft -> {
+                        if (!hasSameFields(oldTaxRate, taxRateDraft)) {
+                            return singletonList(ReplaceTaxRate.of(oldTaxRate.getId(), taxRateDraft));
+                        } else {
+                            return new ArrayList<UpdateAction<TaxCategory>>();
+                        }
+                    })
+                    .orElseGet(() -> singletonList(RemoveTaxRate.of(oldTaxRate.getId())));
+            })
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Checks if there are any new tax rate drafts which are not existing in the {@code oldTaxRates}.
+     * If there are, then "add" tax rate update actions are built.
+     * Otherwise, if there are no new tax rates, then an empty list is returned.
+     *
+     * @param oldTaxRates      the list of old {@link TaxRate}s.
+     * @param newTaxRateDrafts the list of new {@link TaxRateDraft}s.
+     * @return a list of tax rate update actions if there are new tax rate that should be added.
+     *         Otherwise, if the tax rates are identical, an empty list is returned.
+     */
+    @Nonnull
+    private static List<UpdateAction<TaxCategory>> buildAddTaxRateUpdateActions(
+        @Nonnull final List<TaxRate> oldTaxRates,
+        @Nonnull final List<TaxRateDraft> newTaxRateDrafts) {
+
+        final Map<CountryCode, TaxRate> countryCodeTaxRateMap = oldTaxRates
+            .stream()
+            .collect(toMap(TaxRate::getCountry, Function.identity()));
+
+        return newTaxRateDrafts
+            .stream()
+            .filter(taxRateDraft -> !countryCodeTaxRateMap.containsKey(taxRateDraft.getCountry()))
+            .map(AddTaxRate::of)
+            .collect(Collectors.toList());
+    }
+
+    private static boolean hasSameFields(@Nonnull final TaxRate oldTaxRate, @Nonnull final TaxRateDraft newTaxRate) {
+        return Objects.equals(oldTaxRate.getAmount(), newTaxRate.getAmount())
+            && Objects.equals(oldTaxRate.getName(), newTaxRate.getName())
+            && Objects.equals(oldTaxRate.getState(), newTaxRate.getState())
+            && Objects.equals(oldTaxRate.isIncludedInPrice(), newTaxRate.isIncludedInPrice())
+            && sameSubRates(oldTaxRate.getSubRates().stream().filter(Objects::nonNull).collect(toList()),
+            newTaxRate.getSubRates().stream().filter(Objects::nonNull).collect(toList()));
+    }
+
+    private static boolean sameSubRates(final List<SubRate> oldSubRates, final List<SubRate> newSubRates) {
+        boolean same = false;
+
+        if (Objects.nonNull(oldSubRates) && Objects.nonNull(newSubRates)) {
+            if (oldSubRates.size() == newSubRates.size()) {
+                Map<String, SubRate> osr = oldSubRates.stream().collect(toMap(SubRate::getName, subRate -> subRate));
+                Map<String, SubRate> nsr = newSubRates.stream().collect(toMap(SubRate::getName, subRate -> subRate));
```

Comment:
Is name actually unique for the API?

Reply:
I need test this too. Thanks for the review!

Reply:
Tested in impex. When update tax rates, subrate name is allowed duplication. 
https://github.com/commercetools/commercetools-sync-java/pull/524/commits/c052e1aa53227be57e187a14d4ea28b9a9453428

Reply:
@leungkinghin So you match now by a combination of the name and its amount instead?